assets/js/admin/marketplace-suggestions.js

```diff
@@ -0,0 +1,340 @@
+			var visibleSuggestions = [];
+
+			// iterate over all suggestions containers, rendering promos
+			$( '.marketplace-suggestions-container' ).each( function() {
+				// determine the context / placement we're populating
+				var context = this.dataset.marketplaceSuggestionsContext;
+
+				// find promotions that target this context
+				var promos = getRelevantPromotions( marketplaceSuggestionsApiData, context );
+
+				// render the promo content
+				for ( var i in promos ) {
+					var linkText = promos[ i ]['link-text'];
+					var linkoutIsButton = false;
+					if ( promos[ i ]['button-text'] ) {
+						linkText = promos[ i ]['button-text'];
+						linkoutIsButton = true;
+					}
+
+					// dismiss is allowed by default
+					var allowDismiss = true;
+					if ( promos[ 0 ]['allow-dismiss'] === false ) {
+						allowDismiss = false;
+					}
+
+					var content = renderListItem(
+						promos[ i ].slug,
+						promos[ i ].icon,
+						promos[ i ].title,
+						promos[ i ].copy,
+						promos[ i ].url,
+						linkText,
+						linkoutIsButton,
+						allowDismiss
+					);
+					$( this ).append( content );
+					$( this ).addClass( 'showing-suggestion' );
+					visibleSuggestions.push( promos[i].context );
+
+					window.wcSettings.recordEvent( 'marketplace_suggestion_displayed', {
+						suggestionSlug: promos[ i ].slug
+					} );
+				}
+			} );
+
+			// render inline promos in products list
+			if ( 0 === visibleSuggestions.length ) {
+				$( '.wp-admin.admin-bar.edit-php.post-type-product table.wp-list-table.posts tbody').first().each( function() {
+					var context = 'products-list-inline';
+
+					// find promotions that target this context
+					var promos = getRelevantPromotions( marketplaceSuggestionsApiData, context );
+					if ( ! promos || ! promos.length ) {
+						return;
+					}
+
+					// dismiss is allowed by default
+					var allowDismiss = true;
+					if ( promos[ 0 ]['allow-dismiss'] === false ) {
+						allowDismiss = false;
+					}
+
+					// render first promo
+					var content = renderTableBanner(
+						promos[ 0 ].slug,
+						promos[ 0 ].icon,
+						promos[ 0 ].title,
+						promos[ 0 ].copy,
+						promos[ 0 ].url,
+						promos[ 0 ]['button-text'],
+						allowDismiss
+					);
+
+					if ( content ) {
+						// where should we put it in the list?
+						var rows = $( this ).children();
+						var minRow = 3;
+
+						if ( rows.length <= minRow ) {
+							// if small number of rows, append at end
+							$( this ).append( content );
+						}
+						else {
+							// for more rows, append
+							$( rows[ minRow - 1 ] ).after( content );
+						}
+
+						visibleSuggestions.push( context );
+
+						window.wcSettings.recordEvent( 'marketplace_suggestion_displayed', {
+							suggestionSlug: promos[ 0 ].slug
+						} );
+					}
+				} );
+			}
+
+			hidePageElementsForEmptyState( visibleSuggestions );
+		}
+
+		var data =
+		jQuery.getJSON(
+			ajaxurl,
+			{
+				'action': 'marketplace_suggestions',
```

Comment:
nonce?

Reply:
Since this is just a `GET`, and can't change any server-side state (other than transients/cache), there's no need for a nonce (as far as I'm aware).